astroid/brain/brain_typing.py

```diff
@@ -161,10 +137,43 @@ def _looks_like_typing_alias(node: nodes.Call) -> bool:
         isinstance(node, nodes.Call)
         and isinstance(node.func, nodes.Name)
         and node.func.name == "_alias"
-        and isinstance(node.args[0], nodes.Attribute)
+        and (
+            #  _alias function works also for builtins object such as list and dict
+            isinstance(node.args[0], nodes.Attribute)
+            or isinstance(node.args[0], nodes.Name)
+            and node.args[0].name not in ("type",)
+        )
     )
 
 
+def full_raiser(origin_func, attr, *args, **kwargs):
+    """
+    Raises an AttributeInferenceError in case of access to __class_getitem__ method.
+    Otherwise just call origin_func.
+    """
+    if attr == "__class_getitem__":
+        raise AttributeInferenceError("__class_getitem__ access is not allowed")
+    else:
+        return origin_func(attr, *args, **kwargs)
```

Comment:
I would probably put this inside of `_forbid_class_getitem_access`. It's only ever used there.